src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -90,253 +139,3541 @@ static ColliderJntSphInit D_80A65F28 = {
+            }
+            func_800AA000(0.0f, 180, 20, 100);
+            this->flags &= ~ENHORSE_STOPPING_NEIGH_SOUND;
+        } else {
+            EnHorse_StartMountedIdleResetAnim(this);
+        }
+    }
+
+    if (this->skin.skelAnime.curFrame > 29.0f) {
+        this->actor.speedXZ = 0.0f;
+    } else if (this->actor.speedXZ > 3.0f && this->flags & ENHORSE_FORCE_REVERSING) {
+        this->actor.speedXZ = 3.0f;
+    }
+
+    if (SkelAnime_Update(&this->skin.skelAnime)) {
+        if (this->flags & ENHORSE_FORCE_REVERSING) {
+            this->noInputTimer = 100;
+            this->noInputTimerMax = 100;
+            EnHorse_StartReversing(this);
+            this->flags &= ~ENHORSE_FORCE_REVERSING;
+            return;
+        }
+        EnHorse_StartMountedIdleResetAnim(this);
+    }
+}
+
+void EnHorse_StartReversingInterruptable(EnHorse* this) {
+    this->noInputTimerMax = 0;
+    this->noInputTimer = 0;
+    EnHorse_StartReversing(this);
+}
+
+void EnHorse_StartReversing(EnHorse* this) {
+    this->action = ENHORSE_ACT_REVERSE;
+    this->animationIdx = ENHORSE_ANIM_WALK;
+    this->soundTimer = 0;
+    Animation_Change(&this->skin.skelAnime, sAnimationHeaders[this->type][this->animationIdx], 1.0f, 0.0f,
+                     Animation_GetLastFrame(sAnimationHeaders[this->type][this->animationIdx]), ANIMMODE_LOOP, -3.0f);
+}
+
+void EnHorse_Reverse(EnHorse* this, GlobalContext* globalCtx) {
+    f32 stickMag;
+    s16 stickAngle;
+    s16 turnAmount;
+    Player* player = PLAYER;
+
+    EnHorse_PlayWalkingSound(this);
+    EnHorse_StickDirection(&this->curStick, &stickMag, &stickAngle);
+    if (EnHorse_PlayerCanMove(this, globalCtx) == 1) {
+        if (this->noInputTimerMax == 0.0f || (this->noInputTimer > 0.0f && this->noInputTimer < this->noInputTimerMax - 20.0f)) {
+            if (stickMag < 10.0f && this->noInputTimer <= 0.0f) {
+                EnHorse_StartMountedIdleResetAnim(this);
+                this->actor.speedXZ = 0.0f;
+                return;
+            }
+            if (stickMag < 10.0f) {
+                stickAngle = -0x7FFF;
+            } else if (Math_CosS(stickAngle) > -0.5f) {
+                this->noInputTimerMax = 0;
+                EnHorse_StartMountedIdleResetAnim(this);
+                this->actor.speedXZ = 0.0f;
+                return;
+            }
+        } else if (stickMag < 10.0f) {
+            stickAngle = -0x7FFF;
+        }
+    } else if (player->actor.flags & 0x100) {
+        EnHorse_StartMountedIdleResetAnim(this);
+        this->actor.speedXZ = 0.0f;
+        return;
+    } else {
+        stickAngle = -0x7FFF;
+    }
+
+    this->actor.speedXZ = -2.0f;
+    turnAmount = 0x7FFF - stickAngle;
+    turnAmount = CLAMP(turnAmount, -1200.0f, 1200.0f);
+    this->actor.world.rot.y += turnAmount;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+
+    if (0.0f < this->noInputTimer) {
```

Comment:
should be able to flip this around so the constant is on the right. might be relevant in other places